packages/react-swipeable-views/src/SwipeableViews.js

```diff
@@ -744,8 +765,38 @@ class SwipeableViews extends Component {
     } = this.state;
 
     const touchEvents = disabled ? {} : {
-      onTouchStart: this.handleTouchStart,
-      onTouchEnd: this.handleTouchEnd,
+      onTouchStart: (event) => {
+        if (onTouchStart) onTouchStart(event);
+        this.handleTouchStart(event);
+      },
+      onTouchEnd: (event) => {
+        if (onTouchEnd) onTouchEnd(event);
+        this.handleTouchEnd(event);
+      },
+      onMouseDown: (event) => {
+        if (enableMouseEvents) {
+          if (onMouseDown) onMouseDown(event);
+          event.persist();
+          event.touches = [{ pageX: event.pageX, pageY: event.pageY }];
+          this.handleTouchStart(event);
+        }
+      },
+      onMouseUp: (event) => {
+        if (enableMouseEvents) {
+          if (onMouseUp) onMouseUp(event);
+          event.persist();
+          event.touches = [{ pageX: event.pageX, pageY: event.pageY }];
+          this.handleTouchEnd(event);
+        }
+      },
+      onMouseLeave: (event) => {
+        if (enableMouseEvents && this.started) {
+          if (onMouseUp) onMouseUp(event);
```

Comment:
We should be calling `onMouseLeave` no matter if `enableMouseEvents` is enabled or not. Otherwise the event is swallowed for our users.